feat(hw10-bytecodes): implement @Log

ageley opened the pull request.

ageley committed:
feat(hw10-bytecodes): implement @Log

petrelevich commented on hw10-bytecodes/src/main/java/ru/otus/homework/TestLoggingFactory.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Slf4j
+    private static class TestLogging implements TestLoggingInterface {
```
класс, который надо модифицировать, не должен быть внутри класса, создающего прокси.

ageley replied:
Исправил, создал отдельный класс и передаю его экземпляр в метод для проксирования, как параметр.

petrelevich commented on hw10-bytecodes/src/main/java/ru/otus/homework/TestLoggingFactory.java:
```diff
@@ -0,0 +1,59 @@
+    private record LogInvocationHandler(TestLoggingInterface testLogging) implements InvocationHandler {
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+            if (testLogging
```
рефлексивный разбор надо сделать один раз при создании прокси.
Результаты разбора надо сохранить, а потом переиспользовать.

ageley replied:
Исправил, собираю в набор все методы, содержащие аннотацию, при создании прокси-объекта.

ageley committed:
fix(hw10-bytecodes): use reflection in advance

- Parse @Log annotations with reflection in advance
- Pass an object to be proxied as a parameter

ageley committed:
Merge branch 'main' into hw10-bytecodes
